CONTRIBUTING.md

```diff
@@ -0,0 +1,287 @@
+committing:
+
+```bash
+git add -A
+git commit
+```
+
+Format the commit message according to the [Conventional
+Commits](https://www.conventionalcommits.org/en/v1.0.0/) style. For the sanitizer
+example, an appropriate commit title would be:
+
+```
+feat: add text sanitizer
+```
+
+If you need help determining the type of conventional commit for your change, look at
+the history of the file in question with `git log --oneline <filename>` . When you're
+done browsing, press `Q` to exit the interactive log.
+
+> **Tip**
+>
+> With complex changes, you might get stuck choosing a conventional commit type.
+>
+> This may signal that a commit is doing more than one thing and should be broken into
+> multiple smaller commits. A commit should not, for example, refactor code and fix a
+> bug. That should be two separate commits.
+>
+> In other scenarios, multiple types could be appropriate because of the nature of the
+> commit. This can happen with test and docs, which can be used as either types or
+> scopes.
+>
+> Run down the following list and select the highest-ranked type that fits your change:
+>
+> - ci
+> - build
+> - feat
+> - fix
+> - perf
+> - refactor
+> - style
+> - test
+> - docs
+> - chore
+
+Committing triggers the pre-commit hook, which runs the automatic code formatter and the
+fast linters.
+
+If the linters reformatted any of the files, the commit was cancelled. To make the
+changes stick, restage the modified files with `git add -A` and commit again.
+
+### Test the change
+
+Test early and often, especially before you plan to open a pull request.
+
+For low-complexity changes that require basic unit testing, run the fast tests:
+
+```bash
+make test-fast
+```
+
+For complex work, run the full test suite:
+
+```bash
+make test
+```
+
+Running all tests can take a very long time, in some cases an hour.
+
+When iterating and testing, it's a good practice to clean the local temporary files that
+the tests generate:
+
+```bash
+make clean
+```
+
+In rare instances, tests can fail in unpredictable ways, regardless of the state of your
+code. In such cases, it's best to delete your virtual environment and start over:
+
+```bash
+rm -rf ./.venv
+make clean
+make setup
+```
+
+There are also special tests that are feature-specific. Run `make help` to view all of
+them.
+
+### Document the change
+
+Craft CLI has frequent documentation updates to both its user guide and its developer
+material.
+
+Document feature changes and fixes in the [changelog](docs/changelog.rst).
+
+For all documentation updates, build the site locally with:
+
+```bash
+make docs
+```
+
+Use the special documentation linter to check for errors:
+
+```bash
+make lint-docs
+```
+
+### Push the branch and open a PR
+
+Once your work is committed to your branch, push it to your fork:
+
+```bash
+git push -u origin <branch-name>
+```
+
+Finally, [open a PR](https://github.com/canonical/craft-cli/compare) for it on GitHub.
```

Comment:
it's a side-comment but this feels like an "extreme" level of detail to go into. Is that the guideline?